### DISCUSSION TITLE
Streaming to Shoutcast2 fails

### DISCUSSION THREAD

**Discussion by DroidU** [savonet/liquidsoap#3022] [User Support]
**Describe the bug**
When I try to stream to Shoutcast2, I get the following error message:
```
At /usr/share/liquidsoap-lang/libs/icecast.liq, line 36, char 11-16:
    method="icy",

Error 7: Invalid value:
Valid values are: 'source' 'put' or 'post'.
```

**To Reproduce**
```
#!/usr/bin/liquidsoap
s=input.http("http://stream.dancewave.online:8080/dance.ogg")
s=source.on_metadata(s,print)
settings.init.allow_root.set(true)
settings.log.file.path.set("streamtranscoder.log")
settings.log.level.set(6)
settings.init.daemon.pidfile.set(true)
settings.init.daemon.pidfile.path.set("streamtranscoder.pid")
settings.log.stdout.set(true)
output.shoutcast(
  %mp3.cbr(
    bitrate=320,
    samplerate=44100
  ),
  host="172.20.0.9",
  port=2040,
  password="test:#1",
  mksafe(s)
)
```

**Expected behavior**
It should work the same as with Icecast2.

**Version details**

- OS: Debian 12 (bookworm)
- Version 2.2.0+git@2a3bf62e1

**Install method**
Nightly .deb package

**Comment by toots** [comment]
Thank you for reporting. `output.icecast` has been changed to be specific to icecast servers while `output.shoutcast` is now specific to shoutcast servers. You should be able to fix your script changing the operator to `output.shoutcast`. In this case, you shouldn't need to specify the method either.

**Reply by vitoyucepi** [reply]
Hi @toots,
Are you sure about that? I think there's an error here.
https://github.com/savonet/liquidsoap/blob/6028b8671cc62e285f436842b2a3ae3b0966791d/src/libs/icecast.liq#L32-L40

From https://www.liquidsoap.info/doc-dev/reference.html#output.icecast
> method (of type string, which defaults to `"source"`): Method to use with the "http(s)" protocol. One of: `"source"`, `"put"` or `"post"`.

**Reply by vitoyucepi** [reply]
This error happens at parse time.

```ruby
output.shoutcast(
  %mp3,
  host="shoutcast.example.org",
  port=8000,
  password="changeme",
  sine()
)
```

**Reply by toots** [reply]
Woops thanks. This should be fixed in main, will push to `rolling-release-v2.2.x` soon. Thank for insisting.
